flask/app/routes.py

```diff
@@ -162,3 +163,42 @@ def change_password():
     except:
         db.session.rollback()
         return 'Server error', 500
+
+
+@app.route('/api/participants/<participant_id>', methods = ['PATCH'])
+# @login_required # check document for how to test endpoints with login signing? 
+def update_participants(participant_id):
+    patch_json = request.get_json()
+
+    # convert to dictionary
+    patch_d = json.loads(patch_json)
+
+    # get the PK
+    id = patch_d['participant_id']
+
+    # check if PK exists, is this necessary ??
+    q = db.session.query(models.Participant.participant_id)\
+        .filter_by(participant_id = id)
+
+    if q.scalar() == 1:
+        # add time of edit
+        patch_d['updated'] = datetime.now()
+
+        q.update(dict(patch_d)) # for more granular control, i was thinking of looping through
+                                # the dictionary and updating so we prevent the primary key from being changed? 
```

Comment:
Yes, that's the idea. Use the ORM to fetch the corresponding Participant object, update only fields we allow (so no primary key overwrites or changing the dates, for example), commit and then return the object serialized as JSON, or return the error.